virttest/qemu_devices/qdevices.py

```diff
@@ -1143,35 +1143,55 @@ def _cmdline_json(self):
         command_dict = {}
         out = "-%s " % self.type
 
-        usb_storage_driver = self.params.get("driver") == "usb-storage"
+        usb_driver = self.get_param("driver", "").startswith("usb-")
 
-        expect_string_val = ["write-cache", "disable-legacy"]
+        pcic = self.get_param("driver") in ("pcie-root-port")
+
+        pvpanic = self.get_param("driver") in ("pvpanic")
+
+        expect_string_val = ("write-cache", "disable-legacy", "intremap",
+                             "serial")
 
         for key, val in self.params.items():
             # wwn needs to be presented as hexadecimal
-            if key in ["wwn"]:
+            # port from device ( "driver": "pcie-root-port" )
+            if key in ("wwn") or (key == "port" and pcic) \
+                    or (key == "ioport" and pvpanic):
                 command_dict[key] = int(val, 16)
             # physical_block_size from device ("driver": "scsi-hd")
             # logical_block_size from device ("driver": "scsi-hd")
             # bootindex from device ("driver": "scsi-hd")
             # max_sectors from device ("driver": "virtio-scsi-pci")
             # num_queues from device ("driver": "virtio-scsi-pci")
             # virtqueue_size from device ("driver": "virtio-scsi-pci")
-            # discard_granularity from device ("driver": "scsi-hd") and
-            # ("driver": "virtio-blk-pci")
-            elif key in ["physical_block_size", "logical_block_size",
+            # period, max-bytes from device ("driver": "virtio-rng-pci")
```

Comment:
Hi @nickzhq 
Also could help to add these line:
# discard_granularity from device ("driver": "scsi-hd") and
 ("driver": "virtio-blk-pci") ?

Thanks~